pkg/scheduler/factory/plugins.go

```diff
@@ -331,6 +331,13 @@ func RegisterCustomPriorityFunction(policy schedulerapi.PriorityPolicy) string {
 				},
 				Weight: policy.Weight,
 			}
+		} else if policy.Argument.ScarceResourceBinPacking != nil {
+			pcf = &PriorityConfigFactory{
+				MapReduceFunction: func(args PluginFactoryArgs) (algorithm.PriorityMapFunction, algorithm.PriorityReduceFunction) {
+					return priorities.NewScarceResourceBinPacking(policy.Argument.ScarceResourceBinPacking.ScarceResource)
+				},
+				Weight: policy.Weight,
+			}
```

Comment:
curious , is there any ways currently to allow addition of custom priority and predicates without modifying the core k8s ?

Reply:
Yes, there is a way to do that using an extended scheduler, my prototype was built on this. But as many folks were facing the same issue I thought it was a better idea to push this as an optional priority function inside the core scheduler